doc/source/whatsnew/v2.2.0.rst

```diff
@@ -92,6 +92,31 @@ Other API changes
 
 Deprecations
 ~~~~~~~~~~~~
+
+Deprecate alias ``M`` in favour of ``ME`` for offsets
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The alias ``M`` is deprecated in favour of ``ME`` for offsets, please use ``ME`` for "month end" instead of ``M`` (:issue:`9586`)
```

Comment:
we'll probably reword this again before the final release when other offsets are included, so I think this is ok for now